sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java

```diff
@@ -98,9 +101,14 @@ public BeamJoinRel(RelOptCluster cluster, RelTraitSet traits, RelNode left, RelN
       throws Exception {
     BeamRelNode leftRelNode = BeamSqlRelUtils.getBeamRelInput(left);
     BeamRecordSqlType leftRowType = CalciteUtils.toBeamRowType(left.getRowType());
-    PCollection<BeamRecord> leftRows = leftRelNode.buildBeamPipeline(inputPCollections, sqlEnv);
-
     final BeamRelNode rightRelNode = BeamSqlRelUtils.getBeamRelInput(right);
+
+    if (!seekable(leftRelNode, sqlEnv) && seekable(rightRelNode, sqlEnv)) {
```

Comment:
if left side is seekable and right side is not seekable, can we also do a lookup join?

Reply:
Technically yes, but this is a special optiomization rule and I don't want to have too much flexibility. In our development guide it's restricted to be `FACT_TABLE JOIN LOOKUP_TABLE` to make it easy-to-understand.